src/jsonizer/internal/attribute.d

```diff
@@ -33,14 +45,42 @@ struct jsonize {
   }
 }
 
-/// whether to fail deserialization if field is not found in json
-enum JsonizeOptional {
-  unspecified, /// optional status not specified (currently defaults to `no`)
-  no,          /// field is required -- fail deserialization if not found in json
-  yes          /// field is optional -- deserialization can continue if field is not found in json
+/// whether member is required during deserialization
+enum JsonizeIn
+{
+  /// equal yes by default
+  unspecified = 0,
+  /// field is required -- fail deserialization if not found in json
+  yes = 1,
+  /// field is opt -- deserialization can continue if field is not found in json
+  opt = 2,
+  /// newer use json value for deserialization
+  no = 3
+}
+
+/// whether serialized field
+enum JsonizeOut
+{
+  /// equal always by default
+  unspecified = 0,
+  /// always serialize
+  yes = 1,
+  /// serialize only if it not equal initial value of type
+  opt = 2,
+  /// newer serialize field
+  no = 3
+}
+
+/// common in/out using filed
```

Comment:
This doesn't mean much to me. How about 'Shortcut for setting both `JsonizeIn` and `JsonizeOut`'?